client/python/openlineage/client/transport/factory.py

```diff
@@ -0,0 +1,120 @@
+# SPDX-License-Identifier: Apache-2.0.
+import inspect
+import os
+import logging
+from typing import Type, Union, Optional
+
+from openlineage.client.transport.transport import Config, Transport, TransportFactory
+from openlineage.client.utils import try_import_from_string
+
+
+log = logging.getLogger(__name__)
+
+
+try:
+    import yaml
+except ImportError:
+    yaml = False
+
+
+class DefaultTransportFactory(TransportFactory):
+    def __init__(self):
+        self.transports = {}
+
+    def register_transport(self, type: str, clazz: Union[Type[Transport], str]):
+        self.transports[type] = clazz
+
+    def create(self) -> Transport:
+        if yaml:
+            yml_config = self._try_config_from_yaml()
+            if yml_config:
+                return self._create_transport(yml_config)
+        # Fallback to setting HTTP transport from env variables
+        http = self._try_http_from_env_config()
+        if http:
+            return http
+        # If there is no HTTP transport, log events to console
+        from openlineage.client.transport.console import ConsoleTransport, ConsoleConfig
+        return ConsoleTransport(ConsoleConfig())
```

Comment:
I'd add a log statement here just to inform users why events are being printed rather than being sent somewhere

Reply:
Done.